networkx/linalg/laplacianmatrix.py

```diff
@@ -51,12 +54,143 @@ def laplacian_matrix(G, nodelist=None, weight="weight"):
 
     if nodelist is None:
         nodelist = list(G)
-    A = nx.to_scipy_sparse_matrix(G, nodelist=nodelist, weight=weight, format="csr")
+    A = nx.to_scipy_sparse_matrix(G, nodelist=sorted(nodelist), weight=weight, format="csr")
     n, m = A.shape
     diags = A.sum(axis=1)
     D = sp.sparse.spdiags(diags.flatten(), [0], m, n, format="csr")
     return D - A
 
+@not_implemented_for("directed")
+def distance_matrix(G,nodelist=None, weight="weight"):
+    
+    """
+    Notes
+    --------
+    A distance between two nodes is the length of the shortest path between these nodes.
+    Is 0 if is between node with itself and infinity if there is no path joining these two nodes
+    (In case of disconected graph).
+    
+    This is a distance function. Given a graph, it return a pairwise distance matrix
+    between nodes. The graph structure is created through networkx library.
+    
+    
+    Parameters
+    ----------
+    G: Graph (Undirected)
```

Comment:
This function (distance_matrix) is probably not needed. By that I mean, it combines `all_pairs_shortest_path_length` which returns a dict-of-dict structure of lengths keyed by 2 nodes, and then it converts that dict-of-dict as a matrix. The only tricky part is mapping the nodes to the rows and columns.
The following code should do the right thing (I think):
```python
lengths = nx.all_pairs_shortest_path_length(G)
[[length for v, length in unbrs.items()] for u, unbrs in lengths.items()]
```
Or, if it needs a node ordering (which it probably should using a list: `nodelist`)
```python
lengths = nx.all_pairs_shortest_path_length(G)
A = np.array([[lengths[u][v] for v in nodelist] for u in nodelist])
```